Suppress deprecation warnings for isOptimizationEnabled and standard attributes

cary-ilm opened the pull request:
The methods are deprecated for use in application code, but no reason to pollute the build with warnings about the remaining internal usage.

Also, InputPart::isOptimizationEnabled() presumably should be marked as deprecated, since all it does is call the deprecated InputFile::isOptimizationEnabled()

cary-ilm committed:
Suppress deprecation warnings for isOptimizationEanbled and standard attributes

The methods are deprecated for use in application code, but no reason
to pollute the build with warnings about the remaining internal usage.

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented:
@peterhillman, could you take a look at this?

peterhillman commented:
We should be able to remove testOptimized completely. @kdt3rd is testOptimizedInterleavePatterns also unnecessary now? That was originally there to test functionality of code that is now removed, but is it worth keeping it as a test that verifies that the library correctly unpacks into various arrangements of interleaved frame buffers?

cary-ilm committed:
Remove testOptimized.cpp/h entirely.

The test only validates the deprecated API, so it serves no purpose.

This also marks the `InputPart::isOptimizationEnabled()` as
deprecated, since all that does is call
`InputFile::isOptimizationEnabled()`.

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
remove isOptimizationEnabled() and counts from testInterleavePatterns()

This leaves test test in place since it validates interleave/unpack
coverage, but it removes the use of the deprecated API.

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Merge branch 'main' into deprecation-warnings

peterhillman commented on src/lib/OpenEXR/ImfInputPart.cpp:
```diff
@@ -55,7 +55,21 @@ InputPart::isComplete () const
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+
```
Since `file->isOptimizationEnabled()` is no longer meaningful, both the Part and File versions could be switched to simply `return true`, instead of disabling warnings and calling the deprecated function here, but this is fine for now.

peterhillman reviewed (approved):
Looks good!
